pkg/cli/secrets/provider.go

```diff
@@ -0,0 +1,57 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+)
+
+const (
+	ProviderUnknown = "unknown"
+)
+
+// Provider is an interface that provides a secret.
+type Provider interface {
+	// RetrieveSecret retrieves the secret from the provider.
+	// It does the actual work of retrieving the secret..
+	RetrieveSecret(context.Context) (*Secret, error)
```

Comment:
I think I would have expected a provider to be something "reusable" - that you could create a provider of a certain type and then the retrieve function would have a signature somewhat like `func RetrieveSecret(context.Context, Location) (*Secret, error)` or maybe taking the `Identifier` as argument as I guess that must be available when calling..

Reply:
Hmm yes. I started out like that and now I don't remember why I changed it back. Maybe it felt redundant to use that signature when the `Identifier` had already been provided with the `Location` 🤔 Everything here was made with "CLI flags" in mind so I didn't think of the `Provider` as a service-like thing. But I guess there is nothing here preventing us from elevating the package to a more general `secrets` package in go-common?

Reply:
I reverted to the original thought and added `Location` as a parameter to `RetrieveSecret` making `Provider` behave more like a service.